clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/intermediate/VariationArchive.scala

```diff
@@ -247,60 +239,95 @@ object VariationArchive {
+                scvAccessionId,
                 traitsWithoutContent,
                 relevantMappings,
                 traitCounter,
                 rawTrait
               )
               scvTraits.append(WithContent.attachContent(scvTrait, rawTrait))
+              currentScvTraits.append(scvTrait)
             }
+            val traitSet = SCVTraitSet.fromRawAssertionSet(
+              scvAccessionId,
+              rawTraitSet,
+              currentScvTraits.map(_.id).toArray
+            )
+            relatedVcvTraitSetId =
+              SCV.findRelatedVcvTraitSetId(currentScvTraits, vcvTraitSets)
             scvTraitSets.append(WithContent.attachContent(traitSet, rawTraitSet))
+            clinicalAssertionTraitSetId.append(traitSet.id)
           }
 
+          val clinicalAssertionObservationIds = new mutable.ArrayBuffer[String]()
+
           val observationCounter = new AtomicInteger(0)
           rawScv.extractList("ObservedInList", "ObservedIn").foreach { rawObservation =>
             val observation = SCVObservation(
-              id = s"${scv.id}.${observationCounter.getAndIncrement()}",
-              clinicalAssertionId = scv.id
+              id = s"${scvAccessionId}.${observationCounter.getAndIncrement()}"
             )
             rawObservation.extract("TraitSet").foreach { rawTraitSet =>
-              val traitSet =
-                SCVTraitSet.fromRawObservationSet(observation, rawTraitSet)
               val traitCounter = new AtomicInteger(0)
+              val currentScvTraitIds = new mutable.ArrayBuffer[String]()
               MsgTransformations.popAsArray(rawTraitSet, "Trait").foreach { rawTrait =>
                 val scvTrait = SCVTrait.fromRawTrait(
-                  traitSet,
+                  // the setId is the same as the observation.id when extracting from observations
+                  observation.id,
                   traitsWithoutContent,
                   relevantMappings,
                   traitCounter,
                   rawTrait
                 )
                 scvTraits.append(WithContent.attachContent(scvTrait, rawTrait))
+                currentScvTraitIds.append(scvTrait.id)
               }
+              val traitSet = SCVTraitSet.fromRawObservationSet(
+                observation,
+                rawTraitSet,
+                currentScvTraitIds.toArray
+              )
               scvTraitSets.append(WithContent.attachContent(traitSet, rawTraitSet))
             }
             observations.append(WithContent.attachContent(observation, rawObservation))
+            clinicalAssertionObservationIds.append(observation.id)
           }
 
           submitters.append(submitter)
           submissions.append(submission)
+
           // NOTE: It's important to attach content at the very end, to be sure everything
           // that can be modeled has already been popped out of the raw data.
+
+          val rcvId = SCV.findRelatedRcvId(relatedVcvTraitSetId, rcvs)
```

Comment:
I'm :+1: to having fine-grained helper methods in `SCV`, but 👎 to using them in this class. I'll comment in `SCV` with the method signatures I'd prefer.